### PR TITLE
Add a bunch of missing kdoc on Kotlin DSL public APIs

### PR DESCRIPTION
**Pull request by eskatos**
See #275

### CODE REVIEW

File: subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt [hunk 1]
```diff
@@ -36,18 +36,50 @@ inline fun <reified T : Any> Convention.getPluginByName(name: String): T =
...
+inline fun <reified T : Any> Convention.getPlugin(): T =
     getPlugin(T::class)
 
 
+/**
+ * Locates the plugin convention object with the given type.
+ *
+ * @param conventionType the convention plugin type
+ * @return the convention plugin
+ * @throws [IllegalStateException] when there is no such object contained in this convention, or when there are multiple such objects
```
Comment (JLLeitschuh) [inline subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt]: Are you sure this is `IllegalStateException`? Sounds like it should be an `IllegalArgumentException`.

File: subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt [hunk 2]
```diff
@@ -194,6 +194,9 @@ class DependencyHandlerScope(val dependencies: DependencyHandler) : DependencyHa
     inline operator fun <T : ModuleDependency> Configuration.invoke(dependency: T, dependencyConfiguration: T.() -> Unit): T =
         add(name, dependency, dependencyConfiguration)
 
+    /**
+     * `dependencies` block.
```
Comment (JLLeitschuh) [inline subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt]: This line of documentation isn't all that helpful.

File: subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt (hunk 1 above)
Comment (eskatos) [reply subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt]: Yes, it is, see https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/plugins/Convention.java#L46

Comment (JLLeitschuh) [reply subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt]: Strange, not the exception I would have expected.

**Pull request merged**